numba/cuda/compiler.py

```diff
@@ -74,6 +71,69 @@ def compile_kernel(pyfunc, args, link, debug=False, inline=False,
     return cukern
 
 
+# The default compute capability to target when compiling to PTX. This is
+# chosen as the lowest non-deprecated compute capability in the most recent
+# version of the CUDA toolkit supported (10.2 at present).
+DEFAULT_PTX_CC = (5, 2)
+
+
+@global_compiler_lock
+def compile_ptx(pyfunc, args, debug=False, device=False, fastmath=False,
+                cc=DEFAULT_PTX_CC, opt=True):
+    """Compile a Python function to PTX for a given set of argument types.
+
+    :param pyfunc: The Python function to compile.
+    :param args: A tuple of argument types to compile for.
+    :param debug: Whether to include debug info in the generated PTX.
+    :type debug: bool
+    :param device: Whether to compile a device function. Defaults to ``False``,
+                   to compile global kernel functions.
+    :type device: bool
+    :param fastmath: Whether to enable fast math flags (ftz=1, prec_sqrt=0,
+                     prec_div=, and fma=1)
+    :type fastmath: bool
+    :param cc: Compute capability to compile for, as a tuple ``(MAJOR, MINOR)``.
+               Defaults to ``(5, 2)``.
+    :type cc: tuple
+    :param opt: Enable optimizations. Defaults to ``True``.
+    :type opt: bool
+    :return: (ptx, resty): The PTX code and inferred return type
+    :rtype: tuple
+    """
+    cres = compile_cuda(pyfunc, None, args, debug=debug)
+    resty = cres.signature.return_type
+    if device:
+        llvm_module = cres.library._final_module
+        nvvm.fix_data_layout(llvm_module)
+    else:
+        fname = cres.fndesc.llvm_func_name
+        tgt = cres.target_context
+        lib, kernel = tgt.prepare_cuda_kernel(cres.library, fname,
+                                              cres.signature.args, debug=debug)
+        llvm_module = lib._final_module
+
+    options = {
+        'debug': debug,
+        'fastmath': fastmath,
+    }
+
+    opt = opt and 3 or 0
```

Comment:
I think a `3 if opt else 0` is easier to read.

Reply:
Updated to match your suggestion.